src/alire/alire-toml_index.adb

```diff
@@ -1657,6 +1660,30 @@ package body Alire.TOML_Index is
          end if;
 
          O := Label.Value;
+
+         --  Ensure that, for local origins, they exist and are relative to the
+         --  index location.
+         if O.Kind = Filesystem then
+            declare
+               Path      : constant String := +O.Path;
+               Full_Path : constant String :=
+                             Dirs.Full_Name
+                               (Package_Dir
+                                & GNAT.OS_Lib.Directory_Separator
+                                & Path);
+            begin
+               if Path = "" then
+                  Error (+"Empty path given in local origin");
+               elsif Path (Path'First) = GNAT.OS_Lib.Directory_Separator then
```

Comment:
Right you are. There will be a couple more of these through the codebase though...

Reply:
No doubt. :-) What do you think about adding `gnatcoll-core` as a new dependency to leverage its file name handling?

Reply:
In the past I was reticent to add gnatcoll because it seemed a heavy dependency with some others in tow... I see now that it has been split into parts. Since you know it better, do you think gnatcoll-core will be no trouble in general for the build process? If so I'd say let's go for it.

Reply:
Until end of 2017, GNATCOLL was instead a monolithic repo with non-Ada dependencies, so packaging it was messy. Nowadays, `gnatcoll-core` only contains sources that only depend on GNAT itself. The only subtelty is that you have to pass the appropriate scenario variables depending on the platform (`-XGNATCOLL_OS=unix|windows`, `-XGNATCOLL_MMAP=yes|no`, …). As for now we test `alr` only on GNU/Linux, I guess we don’t need to worry too much about that. :-)

I would be fine to take care of adding this dependency and setting it up properly: would this be okay for you?

Reply:
That's great, much simpler to use. Scenario variables are ok with me, I start to pull my hair when there are autoconf/make/file swapping steps involved. For this PR I will use GNAT.OS_Lib.Is_Absolute_Path, so I'd say that you add it once this one is finished.